lib/api/apiUtils/object/objectLockHelpers.js

```diff
@@ -202,7 +202,10 @@ class ObjectLockInfo {
      * @returns {bool} - True if the retention policy allows the objects data to be modified (overwritten/deleted)
      */
     canModifyObject(hasGovernanceBypass) {
-        return !this.isLocked() || (this.isGovernanceMode() && !!hasGovernanceBypass);
+        // can modify object if object is not locked
+        // cannot modify object in any cases if legal hold is enabled
+        // if not legal hold, can only modify object if bypass governance when locked
+        return !this.legalHold && (!this.isLocked() || (this.isGovernanceMode() && !!hasGovernanceBypass));
     }
```

Comment:
I think we should break this up as the return is getting a bit long.
```suggestion
        if (!this.isLocked()) {
            return true;
        }
        return !this.legalHold && this.isGovernanceMode() && !!hasGovernanceBypass;
```